libs/gltfio/src/ResourceLoader.cpp

```diff
@@ -985,46 +996,115 @@ void ResourceLoader::updateBoundingBoxes(FFilamentAsset* asset) const {
+                }
+                if (attr.type == cgltf_attribute_type_weights && cgltf_num_components(accessor->type) >= skinningAttrSize) {
+                    weights.resize(accessor->count * skinningAttrSize);
+                    cgltf_accessor_unpack_floats(accessor, &weights[0], accessor->count * skinningAttrSize);
+                }
+            }
+        }
+        if (!ignoreBindTransform && skin) {
+            std::vector<size_t> jointIndices(rawJoints.begin(), rawJoints.end());
+            auto primitiveCount = static_cast<size_t>(verts.size() / posAttrSize);
+            for (size_t i = 0; i < primitiveCount; i++) {
+                float3 point(verts[posAttrSize * i], verts[posAttrSize * i + 1], verts[posAttrSize * i + 2]);
+                mat4f tmp = mat4f(0.0f);
+                for (size_t j = 0; j < skinningAttrSize; j++) {
+                    size_t jointIndex = jointIndices[skinningAttrSize * i + j];
+                    float jointWeight = weights[skinningAttrSize * i + j];
+                    Entity jointEntity = skin->joints[jointIndex];
+                    mat4f globalJointTransform = tm.getWorldTransform(tm.getInstance(jointEntity));
+                    mat4f inverseBindMatrix = skin->inverseBindMatrices[jointIndex];
+                    tmp += jointWeight * globalJointTransform * inverseBindMatrix;
+                }
+                for (const auto& inverseGlobalTransform: inverseGlobalTransforms) {
+                    mat4f skinMatrix = inverseGlobalTransform * tmp;
+                    if (!weightNormalized) {
+                        skinMatrix /= skinMatrix[3].w;
+                    }
+                    float3 skinnedPoint = (point.x * skinMatrix[0] + point.y * skinMatrix[1] + point.z * skinMatrix[2] + skinMatrix[3]).xyz;
+                    aabb.min = min(aabb.min, skinnedPoint);
+                    aabb.max = max(aabb.max, skinnedPoint);
                 }
-                break;
             }
         }
         *result = aabb;
     };
 
     // Collect all mesh primitives that we wish to find bounds for.
-    std::vector<cgltf_primitive const*> prims;
+    std::vector<SkinnedPrimitive> skinnedPrimitives;
     for (auto iter : nodeMap) {
-        const cgltf_mesh* mesh = iter.first->mesh;
-        if (mesh) {
-            for (cgltf_size index = 0, nprims = mesh->primitives_count; index < nprims; ++index) {
-                prims.push_back(&mesh->primitives[index]);
+        const Skin* skin = nullptr;
+        if (!ignoreBindTransform) {
+            cgltf_skin* const cgltfSkin = iter.first->skin;
+            if (cgltfSkin) {
+                // importSkins unpacked cgltfSkin into FFilamentInstance::SkinVector bijectively so that
+                // the unpacked Skin can be retrieved given cgltfSkin index
+                int skinIndex = cgltfSkin - pImpl->cgltfSkinBaseAddress;
+                skin = &asset->mSkins[skinIndex];
```

Comment:
I just realized I forgot instanced asset. `asset->mSkins` was not used for instanced asset so skin will always be `nullptr`. It feels very expensive to compute aabb with skin bindTransform considered for each instance. So I implicitly disable it with `pImpl->mIgnoreBindTransform = asset->isInstanced();`. Any suggestions?